### PR TITLE
Fix Issue 18624 - wrong behavior of getSymbolsByUDA if one of the symbols having th…

### PR DESCRIPTION
**Pull request by Spoov**
…e UDA is a function

The last change to getSymbolsByUDA introduced a bug, causing it to ignore some symbols

### CODE REVIEW

File: std/traits.d [hunk 1]
```diff
@@ -8032,6 +8033,21 @@ private template getSymbolsByUDAImpl(alias symbol, alias attribute, names...)
...
+    enum Attr;
+    struct A
+    {
+        @Attr void a();
+        @Attr void a(int n);
+              void b();
+        @Attr void c();
+    }
+
+    static assert(getSymbolsByUDA!(A, Attr).length == 3);
```
Comment (wilzbach) [inline std/traits.d]: Why don't you check the result directly?
(would be more robust)

A simple way would be `getSymbolsByUDA!(A, Attr).stringof == "tuple(a, a, c)"`

**Comment by wilzbach** [comment]
As this is a regression it would be great if you could open an issue for it and reference it in the commit message (`git commit --amend`).

Also it should be rebased to stable:

```
git rebase --onto upstream/stable upstream/master
```

See the message of dlang-bot for details.

**Comment by wilzbach** [comment]
See also https://github.com/dlang/phobos/pull/6292 - that's why linking to the PR is a good idea ...

**Comment by Spoov** [comment]
This is my first Pull Request / Issue here so i didn't tough of doing it. I hope i remember doing it next time :)

**Comment by wilzbach** [comment]
No worries. Nobody is perfect - especially on the first run.
Thanks a lot for your contribution! This will be part of of 2.079.1 - it will be released ~ 1.4.18

**Pull request merged**